props/saber_BC_buttons.h

```diff
@@ -386,31 +1134,321 @@ EFFECT(volmax);     // for maximum volume reached
+    current_blade_ = (current_blade_ + NUM_BLADES - 2) % NUM_BLADES + 1;
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade_);
+  }
+
+  void select() override {
+    // Set the current blade for the ChangeBladeLengthMode before transitioning
+    // Deactivate current mode and push to LED length adjustment mode for the selected blade
+    selected_blade_ = current_blade_;
+    highlighted_blade_.Stop(current_blade_);
+    SPEC::SteppedMode::mode_deactivate();
+    pushMode<typename SPEC::ChangeBladeLengthMode>();
+  }
+
+  void exit() override {
+    highlighted_blade_.Stop(current_blade_);
+    SPEC::MenuBase::exit();
+  }
+};
+#endif  // BC_DUAL_BLADES
+
+
+template<class SPEC>
+struct BCChangeBladeLengthMode : public SPEC::MenuBase {
+  virtual int blade() { return selected_blade_; }
+
+  int steps_per_revolution() override {
+    return 40;  // adjust for sensitivity
+  }
+
+  void mode_activate(bool onreturn) override {
+    int len = prop_GetBladeLength(blade());
+    int maxlen = prop_GetMaxBladeLength(blade());
+    saved_len_ = len;
+    if (len == -1) len = maxlen;
+    this->pos_ = len;
+    prop_SetBladeLength(blade(), maxlen);
+    showlen_.Start(blade());
+    SPEC::SteppedMode::mode_activate(onreturn);
+    // mode::getSL<SPEC>()->SayEditBladeLength();
+    say();
+  }
+
+  void mode_deactivate() {
+    showlen_.Stop(blade());
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayWhole(this->pos_);
+  }
+
+  uint16_t size() override {
+    return prop_GetMaxBladeLength(blade());
+  }
+
+  void select() override {
+    prop_SetBladeLength(blade(), this->pos_);
+    prop_SaveState();
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    prop_SetBladeLength(blade(), saved_len_);
+    SPEC::MenuBase::exit();
+  }
+
+  void update() override {
+    hybrid_font.PlayPolyphonic(&SFX_volup);
+    if (!SFX_volup) SaberBase::sound_length = 0.2;
+    this->say_time_ = Cyclint<uint32_t>(millis()) + (uint32_t)(SaberBase::sound_length * 1000) + 400;
+    if (!this->say_time_) this->say_time_ += 1;
+    this->fadeout(SaberBase::sound_length);
+  }
+
+  int getLength() { return this->pos_; }
+
+  ShowColorSingleBladeTemplate<typename SPEC::ShowLengthStyle> showlen_;
+  int saved_len_;
+};
+
+#endif  // DYNAMIC_BLADE_LENGTH
+
+template<class SPEC>
+struct BCMenuSpec {
+  typedef BCVolumeMode<SPEC> BCVolumeMenu;
+#ifdef DYNAMIC_BLADE_LENGTH
+#ifdef BC_DUAL_BLADES
+  typedef BCSelectBladeMode<SPEC> SelectBladeMode;
+#endif  // BC_DUAL_BLADES
+  typedef mode::ShowLengthStyle<SPEC> ShowLengthStyle;
+  typedef BCChangeBladeLengthMode<SPEC> ChangeBladeLengthMode;
+#endif  // DYNAMIC_BLADE_LENGTH
+  typedef mode::SelectCancelMode SelectCancelMode;
+  typedef mode::SteppedMode<SPEC> SteppedMode;
```

Comment:
You'll need to add SteppedModeBase now...

Reply:
ok yes found that, done just not updated yet.

Reply:
done

Reply:
Am I allowed to resolve small, clear cut-and-dry items like this?